src/components/PostDetails.tsx

```diff
@@ -1,117 +1,141 @@
+  selectedPost,
+  createNewComment,
+  setCreateNewComment = () => {},
+}) => {
+  const [postComments, setPostComments] = useState<Comment[]>([]);
+  const [loader, setLoader] = useState<boolean>(false);
+  const [errorMessage, setErrorMessage] = useState<boolean>(false);
+
+  useEffect(() => {
+    setLoader(true);
+    setErrorMessage(false);
+
+    getCommentsData(selectedPost.id)
+      .then(setPostComments)
+      .catch(() => setErrorMessage(true))
+      .finally(() => setLoader(false));
+  }, [selectedPost]);
+
+  const createNewCommentForm = () => {
+    setCreateNewComment(true);
+  };
+
+  const deleteComment = async (commentId: number) => {
+    const temporaryComments = [...postComments];
+
+    setPostComments((prevComments) => prevComments
+      .filter((comment) => comment.id !== commentId));
+
+    try {
+      await deleteCommentData(commentId);
+    } catch {
+      setPostComments(temporaryComments);
+    }
+  };
+
+  const { id, title, body } = selectedPost;
 
-export const PostDetails: React.FC = () => {
   return (
     <div className="content" data-cy="PostDetails">
       <div className="content" data-cy="PostDetails">
         <div className="block">
-          <h2 data-cy="PostTitle">
-            #18: voluptate et itaque vero tempora molestiae
-          </h2>
-
-          <p data-cy="PostBody">
-            eveniet quo quis
-            laborum totam consequatur non dolor
-            ut et est repudiandae
-            est voluptatem vel debitis et magnam
-          </p>
+          <h2 data-cy="PostTitle">{`#${id}: ${title}`}</h2>
+
+          <p data-cy="PostBody">{body}</p>
         </div>
 
         <div className="block">
-          <Loader />
-
-          <div className="notification is-danger" data-cy="CommentsError">
-            Something went wrong
-          </div>
-
-          <p className="title is-4" data-cy="NoCommentsMessage">
-            No comments yet
-          </p>
+          {loader && <Loader />}
 
-          <p className="title is-4">Comments:</p>
-
-          <article className="message is-small" data-cy="Comment">
-            <div className="message-header">
-              <a href="mailto:misha@mate.academy" data-cy="CommentAuthor">
-                Misha Hrynko
-              </a>
-              <button
-                data-cy="CommentDelete"
-                type="button"
-                className="delete is-small"
-                aria-label="delete"
-              >
-                delete button
-              </button>
+          {errorMessage && (
+            <div className="notification is-danger" data-cy="CommentsError">
+              Something went wrong
             </div>
+          )}
 
-            <div className="message-body" data-cy="CommentBody">
-              Some comment
-            </div>
-          </article>
+          {!loader
+            && !errorMessage
+            && postComments
+            && postComments.length === 0 && (
```

Comment:
```suggestion
            && !postComments.length && (
```
If a mentor points out a mistake, it means that you need to fix ALL occurrences of the mistake, not only in the place where the comment was left.